template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplatePreparationObject.java

```diff
@@ -254,8 +264,18 @@ public Builder withFixInputs(Map<String, Object> fixInputs) {
             return this;
         }
 
+        public Builder withIdentityGroupMapping(Map<String, String> groupMapping) {
+            identityGroupMapping = groupMapping == null ? new HashMap<>() : groupMapping;
```

Comment:
Alternatively, `identityGroupMapping = java.util.Objects.requireNonNullElse​(groupMapping, new HashMap<>());`.

Reply:
I followed the pattern around the newly added code